burn-tensor/src/tensor/api/base.rs

```diff
@@ -304,6 +304,80 @@ where
         self.reshape(shape)
     }
 
+    /// Creates a new tensor with added dimensions of size one inserted at the specified indices.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use burn_tensor::backend::Backend;
+    /// use burn_tensor::{Tensor, Shape};
+    ///
+    /// fn example<B: Backend>() {
+    ///     let device = Default::default();
+    ///     let tensor = Tensor::<B, 3>::ones(Shape::new([3, 4,5]), &device);
+    ///     let tensor: Tensor<B, 5> = tensor.unsqueeze_dims(&[0, 4]);
+    ///     println!("{:?}", tensor.shape());
+    ///     // Shape { dims: [1, 3, 4, 5, 1] }
+    /// }
+    /// ```
+    pub fn unsqueeze_dims<const D2: usize>(self, dims: &[isize]) -> Tensor<B, D2, K> {
+        let mut new_dims = [1; D2];
```

Comment:
I feel like this can be implemented in a simpler way:

```rust
let new_dims = [1; D2];
let mut counter = 0;
for i in 0..D2 {
    if !dims.contains(i) {
        new_dims = old_dims[counter];
        counter += 1;
    }
}
tensor.reshape(new_dims);
```

No allocation, and `contains` is extremely fast on small slices (less than 10 elems).

Reply:
I could avoid the allocation by making the second argument `&'op mut [isize]`, and then mutate the new dims in place.

I just realized that the dims can also contain duplicates. The Onnx spec never specified that the values had to be unique, so I wrote it so that there could be duplicates. 

essentially the thinking was unsqueeze dims was equivalent to doing a series of single dim unsqueezes, and if you executed those unsqueezes, it wouldn't matter what order the operations happened: 2 values of `-1` would just result in two axes in the resulting tensor at the end